### DISCUSSION TITLE
How can i run this filter in Staxrip?

### DISCUSSION THREAD

**Discussion by darkcity87** [staxrip/staxrip#718] [Q&A] [answered]
I want to test this filter and use some case how can i run this in staxrip?

[Hysteria](https://github.com/Irrational-Encoding-Wizardry/vapoursynth-Hysteria/blob/master/hysteria.py)

**Comment by JJKylee** [comment, marked as answer]
1. Click `raw` and save `hysteria.py` on your computer.
2. On StaxRip, click _AVS Filters_ > _Filter Setup_ > _VapourSynth_ to go to the VapourSynth mode.
3. Click _Tools_ > _Folders_ > _Plugins_.
4. Copy and paste `hysteria.py` into that folder.
5. Click _VS Filters_ > _Profiles..._.
6. Somewhere under the **[Line]** section, put these lines (empty lines above and beneath should be preserved):

```

Line Darkening | Hysteria =
    # Requires YV12 input, frame-based only
    import hysteria
    clip = hysteria.Hysteria(clip, strength=1.0, usemask=True, lowthresh=6, highthresh=20, showmask=False)

```

7. Load a source file, deinterlace if interlaced, convert to YV12 if not, apply _Line_ > _Line Darkening_ > _Hysteria_. Adjust strength to your taste.

**Reply by JJKylee** [reply]
While replying to this question, I've found an issue.

`%plugin_dir%` is expanded to something like `D:\Tmp\StaxRip-v2.6.0-x64\Settings\Plugins\VapourSynth\` and auto appended, but Python 3.9 returns an error when the path ends with a backslash(`\`) in `sys.path.append(r"~")`.

That is,

`sys.path.append(r"D:\Tmp\StaxRip-v2.6.0-x64\Settings\Plugins\VapourSynth\")` ❌
`sys.path.append(r"D:\Tmp\StaxRip-v2.6.0-x64\Settings\Plugins\VapourSynth")` ✔

As a result, the above 'solution' yields an error like this:

![StaxRip_2 6 0_VS_path_backslash_issue_20210524](https://user-images.githubusercontent.com/3157518/119387972-8ba9e300-bc7e-11eb-9d68-60d9e9b4c8e9.png)

But if I force append the path _without a backslash_,

![StaxRip_2 6 0_VS_path_backslash_issue_no-backslash_20210524](https://user-images.githubusercontent.com/3157518/119388217-d1ff4200-bc7e-11eb-9730-fa5480e54f1f.png)

the error is gone.

Maybe this is a bug with Python 3.9, or VapourSynth R53. 🤔

**Reply by JJKylee** [reply]
### [Updated Solution]

1. Click `raw` and save `hysteria.py` on your computer.
2. On StaxRip, click _AVS Filters_ > _Filter Setup_ > _VapourSynth_ to go to the VapourSynth mode.
3. Click _Tools_ > _Folders_ > _Plugins_.
4. Make a folder named `Scripts` there and copy/paste `hysteria.py` into that folder.
5. Click _VS Filters_ > _Profiles..._.
6. Somewhere under the _**[Line]**_ section, put these lines (empty lines above and beneath should be preserved):

```

Line Darkening | Hysteria =
    # Requires YV12 input, frame-based only
    sys.path.append(r"%plugin_dir%Scripts")
    import hysteria
    clip = hysteria.Hysteria(clip, strength=1.0, usemask=True, lowthresh=6, highthresh=20, showmask=False)

```

7. Load a source file, deinterlace if interlaced, convert to YV12 if not, apply _Line_ > _Line Darkening_ > _Hysteria_. Adjust strength to your taste.

**Reply by darkcity87** [reply]
@JJKylee 
I applied [Updated Solution] working great thanks.
